nrf_modem/Kconfig

```diff
@@ -11,30 +11,41 @@ config NRF_MODEM
 	select NRFX_IPC if (SOC_SERIES_NRF91X || ZTEST)
 	select REQUIRES_FULL_LIBC
 
+if NRF_MODEM
+
 choice NRF_MODEM_BUILD_STRATEGY
 	prompt "Build strategy"
-	depends on NRF_MODEM
-	default NRF_MODEM_LINK_BINARY_CELLULAR
+
+config NRF_MODEM_LINK_BINARY
+	bool "Link binary"
+
+endchoice # NRF_MODEM_BUILD_STRATEGY
+
+choice
+	prompt "Variant"
+	help
+	  Choose which binary variant to link with: cellular or DECT PHY.
 
 config NRF_MODEM_LINK_BINARY_CELLULAR
```

Comment:
Not sure how backwards-compatible we want to be, though these should be renamed.

Reply:
It's good to not break these; plus these are still correct from an NCS perspective.
Elsewhere, where the library can be built from source, these will automatically select the correct flavour.